.devcontainer/Dockerfile

```diff
@@ -0,0 +1,21 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.223.0/containers/python-3/.devcontainer/base.Dockerfile
+
+# [Choice] Python version (use -bullseye variants on local arm64/Apple Silicon): 3, 3.10, 3.9, 3.8, 3.7, 3.6, 3-bullseye, 3.10-bullseye, 3.9-bullseye, 3.8-bullseye, 3.7-bullseye, 3.6-bullseye, 3-buster, 3.10-buster, 3.9-buster, 3.8-buster, 3.7-buster, 3.6-buster
+ARG VARIANT="3.10-bullseye"
+FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
+
+# [Choice] Node.js version: none, lts/*, 16, 14, 12, 10
+ARG NODE_VERSION="none"
+RUN if [ "${NODE_VERSION}" != "none" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
+
+# [Optional] If your pip requirements rarely change, uncomment this section to add them to the image.
+COPY requirements/dev.txt /tmp/pip-tmp/requirements.txt
```

Comment:
Installing the dev dependencies site-wide instead of in a venv or in ~/.local makes it easy to experiment with upgrades/new dependencies in user scope and only commit to them by adding them to the requirements file and rebuilding the container.

Reply:
I actually think this is far and away the best way to set up devcontainers. Once you work with this style a bit I think it'll become obvious. I guess in the most early stages of a project, you maybe want everything to be installed at user scope, so you can rapidly revise, but I would even set things up this way for a greenfield project for myself these days. I love it.

Reply:
Good to know, thanks! I just tried this workflow for myself and it went super smoothly. (Specifically, I ran `pip install --user -U hypothesis` in the devcontainer to try upgrading to the latest hypothesis, confirmed all the hypothesis tests still pass, and know I can now update the requirements files accordingly.)

Updated the comments about this in 9b770be.